api/cloud-control/v1beta1/redisinstance_types.go

```diff
@@ -211,6 +211,7 @@ type RedisInstanceAzure struct {
 	ReplicasPerPrimary int `json:"replicasPerPrimary,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule=(self.authEnabled == false || self.transitEncryptionEnabled == true), message="authEnabled can only be true if TransitEncryptionEnabled is also true"
```

Comment:
I am not sure this validation rule would work as expected? Eg, authEnabled = false and transitEncryptionEnabled = true will trigger it, but the message will not make sence.

Reply:
It will not trigger it

Reply:
`self.authEnabled`==false will be satisfied